node/src/components/consensus/highway_core/finality_detector/rewards.rs

```diff
@@ -92,7 +78,9 @@ fn compute_rewards_for<C: Context>(
         .zip(state.weights())
         .map(|(quorum, weight)| {
             // If the summit's quorum was not enough to finalize the block, rewards are reduced.
-            let finality_factor = if *quorum > state.total_weight() / 2 + fault_w {
+            // A level-1 summit with quorum  q  has FTT  q - 50%. We consider faulty validators to
+            // be part of the summit, so we need  q + f - 50% > f,  i.e.  q > 50%.
+            let finality_factor = if *quorum > state.total_weight() / 2 {
```

Comment:
So, the issue here is that it wasn't aligned with the finality detector? It counts faulty validators towards any summit.

Reply:
No, the issue is that it required the quorum to be > 50% + f, even though > 50% suffices.
(Unless I made another mistake! But that's why I added the comment: In this method, we don't count faulty validators towards the quorum, so the quorum only needs to be > 50%.)